test/Microsoft.AspNetCore.Mvc.Core.Test/Builder/MvcAreaRouteBuilderExtensionsTest.cs

```diff
@@ -33,7 +33,7 @@ public void MapAreaRoute_Simple()
                 kvp =>
                 {
                     Assert.Equal(kvp.Key, "area");
-                    Assert.IsType<RegexRouteConstraint>(kvp.Value);
+                    Assert.IsType<StringRouteConstraint>(kvp.Value);
```

Comment:
this check is present on StringRouteConstraint test in routing repository, i think at this point we do check if the right constraint is used.

i'll do it if you want